samples/RhythmGame/src/main/java/com/donturner/rhythmgame/GameSurfaceView.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.donturner.rhythmgame;
+
+import android.content.Context;
+import android.opengl.GLSurfaceView;
+import android.util.AttributeSet;
+import android.view.MotionEvent;
+import android.view.SurfaceHolder;
+
+public class GameSurfaceView extends GLSurfaceView {
+
+    public static native void native_onTouchInput(int eventType, long timeSinceBootMs, int pixel_x, int pixel_y);
+    public static native void native_surfaceDestroyed();
+    private final RendererWrapper mRenderer;
+
+    public GameSurfaceView(Context context) {
+        super(context);
+
+        // Create an OpenGL ES context
+        /*if(BuildConfig.FLAVOR.equals("GL_3"))
+            setEGLContextClientVersion(3);
+        else if(BuildConfig.FLAVOR.equals("GL_3_2"))
+            setEGLContextClientVersion(3);
+        else*/
+        setEGLContextClientVersion(2);
+        mRenderer = new RendererWrapper();
+        // Set the Renderer for drawing on the GLSurfaceView
+        setRenderer(mRenderer);
+    }
+
+    public GameSurfaceView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+
+        // Create an OpenGL ES context
+        /*if(BuildConfig.FLAVOR.equals("GL_3"))
+            setEGLContextClientVersion(3);
+        else if(BuildConfig.FLAVOR.equals("GL_3_2"))
+            setEGLContextClientVersion(3);
+        else*/
+        setEGLContextClientVersion(2);
+        mRenderer = new RendererWrapper();
+        // Set the Renderer for drawing on the GLSurfaceView
+        setRenderer(mRenderer);
+    }
+
+    @Override
+    public void surfaceDestroyed(SurfaceHolder holder) {
+        super.surfaceDestroyed(holder);
```

Comment:
Does it make sense to run our code first, and then the base class code in this case?

Reply:
Yes, it probably does. Changed.